features/readwrite-splitting/core/src/main/java/org/apache/shardingsphere/readwritesplitting/algorithm/loadbalance/WeightReadQueryLoadBalanceAlgorithm.java

```diff
@@ -50,6 +51,17 @@ public final class WeightReadQueryLoadBalanceAlgorithm implements ReadQueryLoadB
     public void init(final Properties props) {
         this.props = props;
         dataSourceNames = props.stringPropertyNames();
+        ShardingSpherePreconditions.checkState(!dataSourceNames.isEmpty(), () -> new ReadQueryLoadBalanceAlgorithmInitializationExcpetion(getType(), "data source shouldn't be empty!"));
+        for (String dataSourceName : dataSourceNames) {
+            Object weightObject = props.get(dataSourceName);
+            ShardingSpherePreconditions.checkNotNull(weightObject,
+                    () -> new MissingRequiredReadDatabaseWeightException(getType(), String.format("Read database `%s` access weight is not configured.", dataSourceName)));
+            try {
+                Double.parseDouble(weightObject.toString());
+            } catch (final NumberFormatException ex) {
+                throw new InvalidReadDatabaseWeightException(weightObject);
+            }
+        }
```

Comment:
After the check is passed here, the check in `getWeightValue` is unnecessary and can be removed

Reply:
Thanks for your suggestions. Already modified.